index.js

```diff
@@ -189,6 +211,24 @@ module.exports = function validatePeerDependencies(parentRoot, options = {}) {
   }
 };
 
+let AssumptionMapName = '__ValidatePeerDependenciesAssumeProvided';
+if (!(AssumptionMapName in global)) {
+  global[AssumptionMapName] = new Map();
+}
+
+// make sure to re-use the map created by a different instance of
+// validate-peer-dependencies
+let AssumptionMap = global[AssumptionMapName];
+
+module.exports.assumeProvided = function ({ name, version } = {}) {
+  if (name === undefined || version === undefined) {
+    throw new Error(
+      `assumeProvided({ name, version}): name and version are required, but name='${name}' version='${version}'`
+    );
+  }
+  AssumptionMap.set(name, version);
```

Comment:
What happens if you call `assumeProvided` more than once with different versions? Should we error? Clobber - most recent call wins? Clobber - highest number wins? Something else?

Reply:
Discussed this with @hjdivad in DM's, basically this is an escape hatch type of API. Adding an explicit error here would ruin some amount of the "escape hatchness" of it. The simplest implementation is "last write wins", and that is what is implemented (and tested), which seems good enough for now. We can reevaluate if we have other use cases...

Reply:
see the test `it('uses the last value provided'`, i.e. last write wins